### PR TITLE
ENH skip api

### PR DESCRIPTION
**Pull request by tomMoral**
- Add skip API to avoid long runners.
- Rename `scale` to `dimension`
- Add some test for the runner.

### CODE REVIEW

**Comment by agramfort** [comment]
No skip method ? 

Can you also update the narrative doc ?

**Comment by tomMoral** [comment]
There is a skip method.

I will update the doc when I get the time :)
Now is :beer:  time.

File: benchopt/base.py [hunk 1]
```diff
@@ -48,10 +50,30 @@ class BaseSolver(ParametrizedNameMixin, DependenciesMixin, ABC):
...
+        Parameters
+        ----------
+        objective: benchopt.BaseObjective
+            the objective function for the current optimization problem.
+
+        Returns
+        -------
+        skip : bool
+            Wether or not the solver should be skipped.
+        reason : str
```
Comment (agramfort) [inline benchopt/base.py:64]:
```suggestion
        reason : str | None
```

**Review by agramfort — COMMENTED** [review]
@tomMoral can you merge with master, update what's new page and document the skip method in the how.rst ? thanks

**Review by agramfort — APPROVED** [review]
Great let’s move on

**Pull request merged**